include/etcpal/cpp/lock.h

```diff
@@ -0,0 +1,429 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************
+ * This file is a part of EtcPal. For more information, go to:
+ * https://github.com/ETCLabs/EtcPal
+ ******************************************************************************/
+
+/// \file etcpal/cpp/lock.h
+/// \brief C++ wrapper and utilities for etcpal/lock.h
+
+#ifndef ETCPAL_CPP_LOCK_H_
+#define ETCPAL_CPP_LOCK_H_
+
+#include <stdexcept>
+#include "etcpal/lock.h"
+
+namespace etcpal
+{
+/// \defgroup etcpal_cpp_lock etcpal_cpp_lock
+/// \ingroup etcpal_cpp
+/// \brief C++ utilities for the \ref etcpal_lock module.
+
+/// \ingroup etcpal_cpp_lock
+/// \brief A wrapper class for the EtcPal mutex type.
+///
+/// See \ref etcpal_mutex for more information.
+class Mutex
+{
+public:
+  Mutex();
+  ~Mutex();
+
+  Mutex(const Mutex& other) = delete;
```

Comment:
Hmm, I agree rule of 5 means the move functions need to be defined here, but should they be deleted? Moving a mutex seems fine to me.

Reply:
I'm not sure. std::mutex is not moveable or copyable, which I think makes sense. However the scoped wrappers for mutex are moveable. I think that is reasonable (although seems to be an unlikely use case). What do you think?

Reply:
Having no strong opinions of my own, and since the closest analogue to this class is `std::mutex`, I think I'll just mimic that and delete the move functions.